src/wallet-pay-eth.js

```diff
@@ -14,11 +14,34 @@
 
 'use strict'
 const { EvmPay } = require('lib-wallet-pay-evm')
-const { GasCurrency } = require('lib-wallet-util-evm')
+const { GasCurrencyBase } = require('lib-wallet-util-evm')
 const KM = require('./wallet-key-eth.js')
 
+class Ethereum extends GasCurrencyBase {
+  constructor () {
+    const opts = arguments[2] || {}
+    opts.name = opts.name ||  'ETH'
+    opts.base_name = opts.base_name || 'WEI'
+    opts.decimal_places = opts.decimal_places || 18
+    super(...arguments)
```

Comment:
shouldn't this be called with opts after assigning all these default values? `super(...opts)` ?